Examples/Detectors/MuonSpectrometerMockupDetector/src/MockupSectorBuilder.cpp

```diff
@@ -301,14 +356,77 @@ ActsExamples::MockupSectorBuilder::buildSector(
   auto detectorVolume = Acts::Experimental::DetectorVolumeFactory::construct(
       Acts::Experimental::defaultPortalAndSubPortalGenerator(), gctx,
       "detectorVolumeSector", transform,
-      std::move(cylinderVolumesBoundsOfMother),
-      std::vector<std::shared_ptr<Acts::Surface>>{},
+      std::move(cylinderVolumesBoundsOfMother), {},
       detectorCylinderVolumesOfSector, Acts::Experimental::tryAllSubVolumes(),
       Acts::Experimental::tryAllPortalsAndSurfaces());
 
   return detectorVolume;
 }
 
+std::shared_ptr<Acts::Experimental::DetectorVolume>
+ActsExamples::MockupSectorBuilder::buildMultiLayer(
+    std::vector<std::shared_ptr<Acts::Surface>> strawSurfaces,
+    std::string multilayer_name) {
+  auto context = Acts::GeometryContext();
+
+  std::array<std::pair<float, float>, 3> min_max;
+  std::fill(min_max.begin(), min_max.end(),
+            std::make_pair<float, float>(std::numeric_limits<float>::max(),
+                                         -std::numeric_limits<float>::max()));
+
+  // Find the edge values to create the bounds
+
+  for (auto& surf : strawSurfaces) {
+    min_max[0].first =
+        std::min(min_max[0].first, (float)surf->center(context).x());
+    min_max[0].second =
+        std::max(min_max[0].second, (float)surf->center(context).x());
+
+    min_max[1].first =
+        std::min(min_max[1].first, (float)surf->center(context).y());
+    min_max[1].second =
+        std::max(min_max[1].second, (float)surf->center(context).y());
+
+    min_max[2].first =
+        std::min(min_max[2].first, (float)surf->center(context).z());
+    min_max[2].second =
+        std::max(min_max[2].second, (float)surf->center(context).z());
+  }
```

Comment:
All this calculation is what is done already in the construction of the `MultiWireLayerImpl`. Is there a way to pass some of the evaluated quantities, or create here the grid that we can pass to the `NavigationDelegate`?

Reply:
> All this calculation is what is done already in the construction of the `MultiWireLayerImpl`. Is there a way to pass some of the evaluated quantities, or create here the grid that we can pass to the `NavigationDelegate`?

Yes. Actually now it came to my mind that I don't need to do all the calculations for the grid from the surfaces. I can use the volume's bounds and position in order to define the range for the both directions for the grid. And for the number of bins I can do what you suggested, because now it is fixed to the distance of the centers of the surfaces (that's why the sqrt(3) coefficient) and the diameter. But I will try to create the grid here and pass it to the navigation delegate, as you suggest.